src/lv_objx/lv_cpicker.c

```diff
@@ -772,8 +740,39 @@ static void draw_indic(lv_obj_t * cpicker, const lv_area_t * mask, lv_opa_t opa_
         style_cir.body.grad_color = style_cir.body.main_color;
     }
 
+    lv_draw_rect(&indic_area, mask, &style_cir, opa_scale);
+}
+
+static void invalidate_indic(lv_obj_t * cpicker)
+{
+    lv_cpicker_ext_t * ext = lv_obj_get_ext_attr(cpicker);
+    const lv_style_t * style_main = lv_cpicker_get_style(cpicker, LV_CPICKER_STYLE_MAIN);
+    const lv_style_t * style_indic = lv_cpicker_get_style(cpicker, LV_CPICKER_STYLE_INDICATOR);
 
-    lv_draw_rect(&ind_area, mask, &style_cir, opa_scale);
+    lv_area_t indic_area = get_indic_area(cpicker, style_main, style_indic, ext);
+
+    lv_inv_area(lv_obj_get_disp(cpicker), &indic_area);
+}
+
+static lv_area_t get_indic_area(lv_obj_t * cpicker,
+                                const lv_style_t * style_main,
+                                const lv_style_t * style_indic,
+                                const lv_cpicker_ext_t * ext)
+{
```

Comment:
I'd prefer to pass only `cpicker`. It should be very fast to get the `styles` and `ext`.

Reply:
Fixed